stock_analytic/migrations/14.0.2.0.0/pre-migrate.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+
+def _fill_in_move_line(env):
+    analytic_field = [
+        (
+            "analytic_account_id",
+            "stock.move.line",
+            "stock_move_line",
+            "many2one",
+            "int4",
+            "stock_analytic",
+        )
+    ]
+    openupgrade.add_fields(env, analytic_field)
+    query = """
+        UPDATE stock_move_line
+            SET analytic_account_id = sm.analytic_account_id
```

Comment:
Finally, one question. Is the analytic_account_id existing before update the module ?

Reply:
@rousseldenis Event it's merge, could you clarify ?

Reply:
Yes, field is created before pre-migrate.

Reply:
@adrienpeiffer You want to say, maybe a condition is missing.

```suggestion
            SET analytic_account_id = sm.analytic_account_id
            WHERE analytic_account_id IS NULL
```